### PR TITLE
Prepare statement with parameters types

### PR DESCRIPTION
**Pull request by vietj**
Implements https://github.com/eclipse-vertx/vertx-sql-client/issues/700

### CODE REVIEW

**Comment by vietj** [comment]
@BillyYccc can you have a look ?

**Comment by vietj** [comment]
it's currently missing type binding and tests (i.e for now it only does for String to VARCHAR and Integer to INT4)

File: vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/Connection.java [hunk 1]
```diff
@@ -23,10 +23,14 @@
 
 public interface Connection extends CommandScheduler  {
 
+  default boolean isLazyException(Throwable t) {
```
Comment (BillyYccc) [inline vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/Connection.java:26]: what does lazy exception mean?

Comment (vietj) [reply vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/Connection.java:26]: it means it checks whether the exception is a lazy statement, I'll try finding a better name, for now this PR is not polished at all but it provides most of what we want to do.

Comment (vietj) [reply vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/Connection.java:26]: something like isIncompletePreparedStatementException(T) would be better likely

**Pull request merged**